crates/ingress-dispatcher/src/dispatcher.rs

```diff
@@ -42,9 +51,18 @@ pub trait DispatchIngressRequest {
 #[derive(Default)]
 struct IngressDispatcherState {
     msg_index: AtomicU64,
+
     // This map can be unbounded, because we enforce concurrency limits in the ingress
     // services using the global semaphore
-    waiting_responses: DashMap<IngressCorrelationId, IngressResponseSender>,
+    waiting_responses: DashMap<
+        (InvocationIdOrIdempotencyId, RequestType),
+        HashMap<IngressResponseWaiterId, IngressInvocationResponseSender>,
+    >,
```

Comment:
So long story short with this implementation we won't be able to achieve linearizable reads because the result of a read might originate from before the read request was sent.